src/routers/user.router.ts

```diff
@@ -16,45 +17,84 @@ userRouter.get("", async (req: Request, res: Response) => {
   }
 });
 
-userRouter.get("/:id", async (req: Request, res: Response) => {
-  try {
-    const id: number = await parseInt(req.params.id, 10);
-    const user = await userV2Service.getUserById(id);
-    if (user != undefined) {
-      return res.status(200).json(user);
-    } else {
-      return res.status(404).send(errorHandling(undefined));
+userRouter.get(
+  "/:id",
+  param("id").exists().isNumeric(),
+  async (req: Request, res: Response) => {
+    const errors = validationResult(req);
+    if (!errors.isEmpty()) {
+      return res.status(400).json({ errors: errors.array() });
+    }
+    try {
+      const id: number = await parseInt(req.params.id, 10);
+      const user = await userV2Service.getUserById(id);
+      if (user != undefined) {
+        return res.status(200).json(user);
+      } else {
+        return res.status(404).send(errorHandling(undefined));
+      }
+    } catch (err) {
+      res.status(401).json(err);
     }
-  } catch (err) {
-    res.status(401).json(err);
   }
-});
-userRouter.post("", async (req: Request, res: Response) => {
-  try {
-    const newUser = await userV2Service.save(req.body);
-    res.status(201).json(newUser);
-  } catch (err) {
-    res.status(401).json(err);
+);
+userRouter.post(
+  "",
+  body("name").exists().isLength({ max: 45 }),
+  body("email").exists().isEmail(),
+  body("password").isLength({ min: 6, max: 6 }),
```

Comment:
Limiting to max 6 characters a password?  No go.
Users should have the possibility of creating a password with up to 20 characters (at least)

Reply:
ok will change it thanks